examples/react-hooks-example/src/post.md

```diff
@@ -0,0 +1,221 @@
+        pad={
+            {
+                left: 'medium',
+                right: 'small',
+                vertical: 'small'
+            }
+        }
+        elevation='medium'
+        style={
+            {zIndex: '1'}
+        }
+        {...props}/>
+);
+```
+
+Now we've got the skeleton of an app.  Let's wire it up to Ethereum.
+
+## Setting up the plumbing
+
+First, we need a web3 provider.  For simplicity's sake, we're going to take advantage of the openly accessible Ethereum gateway provided by the fine folks over at [Cloudflare](https://developers.cloudflare.com/distributed-web/ethereum-gateway/).
+```const web3 = new Web3('https://cloudflare-eth.com');```
+
+Next, we need a smart contract to interact with.  The ABI for the Uniswap Exchange contract.  You can get it [here](https://raw.githubusercontent.com/acolytec3/subspace-demo/master/src/contract/exchange_abi.json).  I put it in a subdirectory called `contract` but do what you will.  Then, just import it like any other dependency. 
+`import exchangeABI from './contract/exchange_abi.json'`
+
+Finally, let's set up Subspace and the smart contract we're going to be working with:
+```js
+const subspace = new Subspace(web3.currentProvider);
+var dai = new web3.eth.Contract(exchangeABI, '0x2a1530C4C41db0B0b2bB646CB5Eb1A67b7158667');
+const daiContract = subspace.contract(dai);
+subspace.init();
+```
+ Now, we're reading to start doing something interesting.
+
+## Getting the data
+
+While there are any number of ways to do this, we're going to use React Hooks to manage our Subspace observables that pipe date to the front end.
+
+The basic premise of this Dapp is that we're going to show the exchange rate on most recent 5 DAI to ETH trades and then compute an average. I'm going to set up 2 Subspace observables, one to stream the trades from the Uniswap exchange contract and then another to package that up into an array of the last 5 trades to put in a widget in the Dapp.
+
+### Setting up the observables
+
+We're going to use State and Effect Hooks to manage the observables and the data they return.  As a reminder, [State hooks](https://reactjs.org/docs/hooks-state.html) are a React concept for managing state in functional components.  Here's the declaration of the State Hooks.
+
+```js
+    const [txnObserver, setObservable] = useState();
+    const [last5Observer, setLast5Observer] = useState();
+    const [latestBlock, setBlock] = useState();
+    const [last5, setLast5] = useState([]);
+```
+
+Next, we're going to set up three [Effect Hooks](https://reactjs.org/docs/hooks-effect.html) to define and then manage the subscriptions for the Subspace observables.  A full explanation of how Effect Hooks work is beyond the scope of this article but the basic point to remember here is that these hooks are intended to be side effects of the rendering process in the `App` component and each effect should be isolated from the others or you will get odd behavior.
+
+### Observable declaration
```

Comment:
Maybe this would be better if it was above the previous paragraph (about Event Hooks)?

Reply:
Got it, reworked slightly in next commit.